devel/boost/files/patch-boost-python3.diff

```diff
@@ -3,14 +3,25 @@
 @@ -13,6 +13,10 @@
  #include <boost/mpi/python/serialize.hpp>
  #include <boost/mpi.hpp>
- 
+
 +#if PY_MAJOR_VERSION >= 3
 +#define PyInt_Type PyLong_Type
 +#endif
 +
  namespace boost { namespace mpi { namespace python {
- 
+
  void export_datatypes()
+--- libs/python/src/converter/builtin_converters.cpp.orig	2017-12-13 17:56:47.000000000 -0600
++++ libs/python/src/converter/builtin_converters.cpp	2018-07-05 22:46:05.000000000 -0500
+@@ -48,7 +48,7 @@
+ #else
+   void* convert_to_cstring(PyObject* obj)
+   {
+-      return PyUnicode_Check(obj) ? _PyUnicode_AsString(obj) : 0;
++      return (void*)(PyUnicode_Check(obj) ? _PyUnicode_AsString(obj) : 0);
```

Comment:
IMO it's better to pick the upstream commit https://github.com/boostorg/python/commit/660487c43fde76f3e64f1cb2e644500da92fe582. Note that this commit is included in boost 1.67 (https://trac.macports.org/ticket/56294).

Reply:
👍 That is indeed much better. This pull request can be closed.